samples/nrf9160/coap_client/README.rst

```diff
@@ -3,12 +3,24 @@
 nRF9160: nRF CoAP Client
 ########################
 
-The nRF CoAP Client sample demonstrates how to receive data from a public CoAP server with an nRF9160 SiP.
+The nRF CoAP Client sample demonstrates the communication between a public CoAP server and an nRF9160 SiP which acts as the CoAP client.
 
 Overview
-*********
+********
 
-The sample connects to a public CoAP test server, sends periodic GET request for a test resource that is available on the server, and prints the data that is received.
+The nRF CoAP Client sample performs the following actions:
+
+#. Connect to the configured public CoAP test server (indicated by the parameter ``CONFIG_COAP_SERVER_HOSTNAME``)
+#. Send periodic GET request for a test resource (indicated by the parameter ``CONFIG_COAP_RESOURCE``) that is available on the server
+#. Display the received data about the resource on a terminal emulator.
```

Comment:
Question for developers: 
Should we explain more about the received data? Any particular CoAP response code or set of codes relevant to be documented? Anything like  the CoAP client output shown in [https://infocenter.nordicsemi.com/index.jsp?topic=%2Fcom.nordic.infocenter.sdk5.v15.3.0%2Fiot_sdk_user_guides_java_coaps.html](url)

Reply:
The sample prints data in the following format:
```
CoAP request sent: token 0x9772
CoAP response: code: 0x45, token 0x9772, payload: 15:39:40
```
It does not print every single CoAP frame content, just the essentials. These are, in the provided sample:
* code: 0x45 - CoAP response code (2.05 Content), this will not change across the response,
* token 0x9781 - CoAP token, this will be unique per request/response pair,
* payload: 15:40:55 - the actual message paylaod. The resource we query in the sample returns current UTC  time.